Hotfix/7.1.1 (#63)
iExecBlockchainComputing/iexec-result-proxy: README.md

# iexec-result-proxy

## Overview

> iExec Result Proxy to persist task execution results to IPFS.

Reviewer: ```suggestion
The iExec Result Proxy stores results of task executions to IPFS.
```

The iExec Result Proxy is available as an OCI image on [dockerhub](https://hub.docker.com/r/iexechub/iexec-result-proxy/tags).

To run properly, the iExec Result Proxy requires:
* A reachable blockchain node URL hosting iExec smart contracts
* A reachable MongoDB instance to persist its data
* A reachable IPFS node where iExec tasks results will be pushed and retrieved

## Configuration

You can configure the result-proxy with the following properties:

> | Environment variables       | Description                                                                                            | Values           | Default value                              |
> |-----------------------------|--------------------------------------------------------------------------------------------------------|------------------|--------------------------------------------|
> | IEXEC_RESULT_PROXY_PORT     | Server port of the result proxy.                                                                       | Positive integer | 13200                                      |
> | MONGO_HOST                  | Host to connect to the MongoDB. It is recommended to use the same as the one hosting the result proxy. | String           | localhost                                  |
> | MONGO_PORT                  | Port to connect to the MongoDB.                                                                        | Positive integer | 13202                                      |
> | IEXEC_CHAIN_ID              | ID of the chain.                                                                                       | Integer          | 17                                         |
> | IEXEC_IS_SIDECHAIN          | Whether the chain node is a sidechain or an ERC20 token.                                               | Boolean          | false                                      |
> | IEXEC_PRIVATE_CHAIN_ADDRESS | Private URL to connect to the blockchain node.                                                         | String           | http://localhost:8545                      |
> | IEXEC_PUBLIC_CHAIN_ADDRESS  | [unused] Public URL to connect to the blockchain node.                                                 | String           | http://localhost:8545                      |
> | IEXEC_HUB_ADDRESS           | Ethereum address to interact with the iExec smart contracts.                                           | String           | 0xBF6B2B07e47326B7c8bfCb4A5460bef9f0Fd2002 |
> | IEXEC_START_BLOCK_NUMBER    | [Unused]                                                                                               | Positive integer | 0                                          |
> | IEXEC_GAS_PRICE_MULTIPLIER  | Transactions will be sent with `networkGasPrice * IEXEC_GAS_PRICE_MULTIPLIER`.                         | Float            | 1.0                                        |
> | IEXEC_GAS_PRICE_CAP         | In Wei, will be used for transactions if `networkGasPrice * IEXEC_GAS_PRICE_MULTIPLIER > gasPriceCap`. | Integer          | 22000000000                                |
> | IEXEC_IPFS_HOST             | Host to connect to the IPFS node.                                                                      | String           | 127.0.0.1                                  |
> | IEXEC_IPFS_PORT             | Server port of the IPFS node.                                                                          | Positive integer | 5001                                       |

Reviewer: ```suggestion
|Environment variables|Description|Values|Default value|
|-|-|-|-|
|IEXEC_RESULT_PROXY_PORT|Server port of the result proxy.|Positive integer|13200|
|MONGO_HOST|Host to connect to the MongoDB. It is recommended to use the same as the one hosting the result proxy.|String|localhost|
|MONGO_PORT|Port to connect to the MongoDB.|Positive integer|13202|
|IEXEC_CHAIN_ID|ID of the chain.|Integer|17|
|IEXEC_IS_SIDECHAIN|Whether the chain node is a sidechain or an ERC20 token.|Boolean|false|
|IEXEC_PRIVATE_CHAIN_ADDRESS|Private URL to connect to the blockchain node.|String|http://localhost:8545|
|IEXEC_PUBLIC_CHAIN_ADDRESS|[unused] Public URL to connect to the blockchain node.|String|http://localhost:8545|
|IEXEC_HUB_ADDRESS|Ethereum address to interact with the iExec smart contracts.|String|0xBF6B2B07e47326B7c8bfCb4A5460bef9f0Fd2002|
|IEXEC_START_BLOCK_NUMBER|[Unused]|Positive integer|0|
|IEXEC_GAS_PRICE_MULTIPLIER|Transactions will be sent with `networkGasPrice * IEXEC_GAS_PRICE_MULTIPLIER`.|Float|1.0|
|IEXEC_GAS_PRICE_CAP|In Wei, will be used for transactions if `networkGasPrice * IEXEC_GAS_PRICE_MULTIPLIER > gasPriceCap`.|Integer|22000000000|
|IEXEC_IPFS_HOST|Host to connect to the IPFS node.|String|127.0.0.1|
|IEXEC_IPFS_PORT|Server port of the IPFS node.|Positive integer|5001|
```

Author: It is intellij auto-formatting of markdown tables.
It triggers itself automatically.
And the preview mode within intellij is reaaly nice to have.

Author: Need to disable Markdown auto-assistance in settings to solve the formatting point

> # iexec-result-proxy

Reviewer: ```suggestion
# iExec Result Proxy
```

## Overview

> The iExec Result Proxy stores results of task executions to IPFS.

Reviewer: ```suggestion
The iExec Result Proxy stores results of iExec tasks on IPFS.
```

Author: See 68f64cc

> The iExec Result Proxy is available as an OCI image on [dockerhub](https://hub.docker.com/r/iexechub/iexec-result-proxy/tags).

Reviewer: ```suggestion
The iExec Result Proxy is available as an OCI image on [Docker Hub](https://hub.docker.com/r/iexechub/iexec-result-proxy/tags).
```

Author: See 58bb787

> The iExec Result Proxy stores results of iExec tasks to IPFS.
> 
> The iExec Result Proxy is available as an OCI image on [Docker Hub](https://hub.docker.com/r/iexechub/iexec-result-proxy/tags).
> 
> To run properly, the iExec Result Proxy requires:
> * A reachable blockchain node URL hosting iExec smart contracts.
> * A reachable MongoDB instance to persist its data.
> * A reachable IPFS node where iExec tasks results will be pushed and retrieved.

Reviewer: ```suggestion
The iExec Result Proxy stores results of iExec tasks on IPFS.

The iExec Result Proxy is available as an OCI image on [Docker Hub](https://hub.docker.com/r/iexechub/iexec-result-proxy/tags).

To run properly, the iExec Result Proxy requires:
* A blockchain node. iExec smart contracts must be deployed on the blockchain network.
* A MongoDB instance to persist its data.
* An IPFS node where iExec tasks results will be pushed and retrieved.
```

## Configuration

> You can configure the result-proxy with the following properties:

Reviewer: ```suggestion
You can configure the iExec Result Proxy with the following properties:
```

> | Environment variables | Description | Values | Default value |
> | --- | --- | --- | --- |
> | IEXEC_RESULT_PROXY_PORT | Server port of the result proxy.  | Positive integer | `13200` |
> | MONGO_HOST | Host to connect to the MongoDB. It is recommended to use the same as the one hosting the result proxy. | String | `localhost` |
> | MONGO_PORT | Port to connect to the MongoDB. | Positive integer | `13202` |
> | IEXEC_CHAIN_ID | ID of the chain on the blockchain node. | Integer | `17` |
> | IEXEC_IS_SIDECHAIN | Whether the chain node is a sidechain or an ERC20 token. | Boolean | `false`  |
> | IEXEC_PRIVATE_CHAIN_ADDRESS | Private URL to connect to the blockchain node. | URL | `http://localhost:8545` |
> | IEXEC_PUBLIC_CHAIN_ADDRESS  | [unused] Public URL to connect to the blockchain node. | URL | `http://localhost:8545` |
> | IEXEC_HUB_ADDRESS | Ethereum address to interact with the iExec smart contracts. | Ethereum address | `0xBF6B2B07e47326B7c8bfCb4A5460bef9f0Fd2002` |
> | IEXEC_START_BLOCK_NUMBER | [Unused] | Positive integer | `0` |
> | IEXEC_GAS_PRICE_MULTIPLIER | Transactions will be sent with `networkGasPrice * IEXEC_GAS_PRICE_MULTIPLIER`. | Float | `1.0` |
> | IEXEC_GAS_PRICE_CAP | In Wei, will be used for transactions if `networkGasPrice * IEXEC_GAS_PRICE_MULTIPLIER > gasPriceCap`. | Integer | `22000000000` |
> | IEXEC_IPFS_HOST | Host to connect to the IPFS node. | String  | `127.0.0.1` |
> | IEXEC_IPFS_PORT | Server port of the IPFS node. | Positive integer | `5001` |

Reviewer: ```suggestion
| Environment variable | Description | Type | Default value |
| --- | --- | --- | --- |
| IEXEC_RESULT_PROXY_PORT | Port of the result proxy node.  | Positive integer | `13200` |
| MONGO_HOST | Host to connect to the MongoDB instance. It is recommended to use the same as the one hosting the result proxy. | String | `localhost` |
| MONGO_PORT | Port to connect to the MongoDB instance. | Positive integer | `13202` |
| IEXEC_CHAIN_ID | Chain ID of the blockchain network to connect. | Integer | `17` |
| IEXEC_IS_SIDECHAIN | Define if iExec on-chain protocol is built on top of token (`false`) or native currency (`true`). | Boolean | `false`  |
| IEXEC_PRIVATE_CHAIN_ADDRESS | Private URL to connect to the blockchain node. | URL | `http://localhost:8545` |
| IEXEC_PUBLIC_CHAIN_ADDRESS  | [unused] Public URL to connect to the blockchain node. | URL | `http://localhost:8545` |
| IEXEC_HUB_ADDRESS | Proxy contract address of the iExec on-chain protocol. | Ethereum address | `0xBF6B2B07e47326B7c8bfCb4A5460bef9f0Fd2002` |
| IEXEC_START_BLOCK_NUMBER | [Unused] | Positive integer | `0` |
| IEXEC_GAS_PRICE_MULTIPLIER | Transactions will be sent with `networkGasPrice * IEXEC_GAS_PRICE_MULTIPLIER`. | Float | `1.0` |
| IEXEC_GAS_PRICE_CAP | In Wei, will be used for transactions if `networkGasPrice * IEXEC_GAS_PRICE_MULTIPLIER > gasPriceCap`. | Integer | `22000000000` |
| IEXEC_IPFS_HOST | Host to connect to the IPFS node. | String  | `127.0.0.1` |
| IEXEC_IPFS_PORT | Port of the IPFS node. | Positive integer | `5001` |
```

Reviewer: I don't understand "It is recommended to use the same as the one hosting the result proxy." (?)

Author: I try to enforce this point;
It is generally recommended to host the database on the same host as the backend it serves

Author: In my proposal, I tried to stick to Spring Boot  **Common Application Properties** documentation
https://docs.spring.io/spring-boot/docs/current/reference/html/application-properties.html
An example: **Server port of the result proxy.** was derived from **Server HTTP port**.

I don't know if we want to follow this or not, we may even go for a straight copy paste for some properties
